### DISCUSSION TITLE
Cannot insert/upsert in a RLS read protected table

### DISCUSSION THREAD

**Discussion by laem** [supabase/supabase#5073] [Questions]
Hi, I'm trying to implemented the type of use case described in #3500, but am unable to do this simple thing. 

My `answers` database has these two policies : 

```
CREATE POLICY update_answers ON answers FOR UPDATE using(true);
CREATE POLICY insert_answers ON answers FOR INSERT with check(true);
```
It has no SELECT pollicy, I don't want users to be able to read this table, apart from a special function that lets users get all the answers but only for a specific `survey`. 

I want the anonymous user to be able to insert and update using the following js code : 

First problem, if I use `upsert`, it doesn't work, says the user has no right for this.

So I tried using two separate blocks, first inserting with "returning: minimal" (it works), storing a cache in the user's browser, then updating. 
```js
database.from('answers').update({ data }).eq('id', cachedSurveyAnswerId))                                                     
```
Then I get the following error : PATCH 404 not found. 

Supabase / postrgre's RLS policies do not seem to work for my non-authenticated but restricted read access use case.

**Comment by GaryAustin1** [comment]
You need to make sure you set the option returning minimal as both insert and update do an extra select to return data by default.
{ returning:'minimal' } needs to be added to your .update call.

**Reply by laem** [reply]
I just tried with the minimal option, but no it doesn't work :/
```js

database                                                             
.from('answers')                                                                                                                                                         
.update(
{ data },
{
returning: 'minimal',
}).eq('id', cachedSurveyId))
```

I'm still getting a XHR 404 not found.

**Reply by GaryAustin1** [reply]
I’d check the id in your network request is what you think it is,
Also what is data?  Not sure you are just saving time here or it is an object. If object drop the extra {} around it…although think that would error versus 404.

**Reply by laem** [reply]
Yes I've checked the id in the devtools :/ 
The data is a json column. 
Thanks for your tips though.

**Reply by GaryAustin1** [reply]
The {data} thing still does not look right.  Update takes an object with col:value pairs.  Is data a variable or are you just not typing your col:val pairs and data means them?
For grins also turn on select RLS and see if update works.

**Reply by laem** [reply]
Yes in modern js `{data}` means `{data: data}`, the second one being an object. Yes my naming is quite confusing :sweat:
